File: ai-interview/frontend/src/types/report.ts
export interface DimensionScore {
  score: number;
  max: number;
  comment: string;
}

export interface QuestionReview {
  question: string;
  answer_summary: string;
  score: number;
  comment: string;
  improved_answer_suggestion: string;
}

export interface Report {
  id?: string;
  interview_id?: string;
  total_score: number;
  dimension_scores: Record<string, DimensionScore>;
  overall_comment: string;
  strengths: string[];
  weaknesses: string[];
  resume_risks: string[];
  question_reviews: QuestionReview[];
  next_training_plan: string[];
}

File: ai-interview/backend/app/schemas/report.py
from pydantic import BaseModel

from app.schemas.ai import DimensionScore, QuestionReview


class ReportOut(BaseModel):
    id: str | None = None
    interview_id: str | None = None
    provider: str | None = None
    model: str | None = None
    total_score: int
    dimension_scores: dict[str, DimensionScore]
    overall_comment: str
    strengths: list[str]
    weaknesses: list[str]
    resume_risks: list[str]
    question_reviews: list[QuestionReview]
    next_training_plan: list[str]

    model_config = {"from_attributes": True}
